modules/compatibility-tests/src/test/java/org/apache/ignite/internal/PersistenceTestBase.java

```diff
@@ -0,0 +1,142 @@
+            + "  network: {\n"
+            + "    port: {},\n"
+            + "    nodeFinder.netClusterNodes: [ {} ]\n"
+            + "  },\n"
+            + "  storage.profiles: {"
+            + "        " + DEFAULT_AIPERSIST_PROFILE_NAME + ".engine: aipersist, "
+            + "        " + DEFAULT_AIMEM_PROFILE_NAME + ".engine: aimem, "
+            + "        " + DEFAULT_ROCKSDB_PROFILE_NAME + ".engine: rocksdb"
+            + "  },\n"
+            + "  clientConnector.port: {},\n"
+            + "  clientConnector.sendServerExceptionStackTraceToClient: true,\n"
+            + "  rest.port: {},\n"
+            + "  failureHandler.dumpThreadsOnFailure: false\n"
+            + "}";
+
+    // If there are no fields annotated with @Parameter, constructor injection will be used, which is incompatible with the
+    // Lifecycle.PER_CLASS.
+    @SuppressWarnings("unused")
+    @Parameter
+    String baseVersion;
+
+    @WorkDirectory
+    private static Path WORK_DIR;
+
+    protected IgniteCluster cluster;
+
+    @SuppressWarnings("unused")
+    @BeforeParameterizedClassInvocation
+    void startCluster(String baseVersion, TestInfo testInfo) {
+        ClusterConfiguration clusterConfiguration = ClusterConfiguration.builder(testInfo, WORK_DIR)
+                .defaultNodeBootstrapConfigTemplate(NODE_BOOTSTRAP_CFG_TEMPLATE)
+                .build();
+
+        int nodesCount = nodesCount();
+
+        cluster = new IgniteCluster(clusterConfiguration);
+        cluster.start(baseVersion, nodesCount);
+
+        cluster.init(this::configureInitParameters);
+
+        try (IgniteClient client = cluster.createClient()) {
+            setupBaseVersion(client);
+        }
+
+        cluster.stop();
+
+        cluster.startEmbedded(nodesCount);
+    }
+
+    @SuppressWarnings("unused")
+    @AfterParameterizedClassInvocation
+    void stopCluster() {
+        if (cluster != null) {
+            cluster.stop();
+        }
+    }
+
+    protected String getNodeBootstrapConfigTemplate() {
+        return NODE_BOOTSTRAP_CFG_TEMPLATE;
+    }
+
+    protected int nodesCount() {
+        return 3;
+    }
+
+    /**
+     * This method can be overridden to add custom init parameters during cluster initialization.
+     */
+    protected void configureInitParameters(InitParametersBuilder builder) {
+    }
+
+    protected abstract void setupBaseVersion(Ignite baseIgnite);
+
+    protected List<List<Object>> sql(String query) {
+        return ClusterPerClassIntegrationTest.sql(cluster.node(0), null, null, null, query);
+    }
+
+    private static List<String> baseVersions() {
+        List<String> versions = IgniteVersions.INSTANCE.versions().stream().map(Version::version).collect(Collectors.toList());
+        if (System.getProperty("testAllVersions") != null) {
+            return versions;
+        } else {
+            // Take at most two latest versions by default.
+            int fromIndex = Math.max(versions.size() - 2, 0);
```

Comment:
Can we make this overridable, so a derived class can decide which versions to use?